src/model/neuroevolution/individual.py

```diff
@@ -0,0 +1,220 @@
+    adaptation: float
+
+    available_reproductions: ClassVar[
+        List[
+            Callable[
+                ["Individual", "Individual"], Tuple[ChildIndividual, ChildIndividual]
+            ]
+        ]
+    ]
+
+    @staticmethod
+    def weight_swap_reproduction(
+        father1: "Individual", father2: "Individual", weights_to_be_swapped: int = 1
+    ) -> Tuple[ChildIndividual, ChildIndividual]:
+        """
+        Return two new individuals (children). They are created by
+        swapping one or more parents's weights.
+
+        :param father1: First parent
+        :param father2: Second parent
+        :param weights_to_be_swapped: number of weights to be swapped
+        :return: Tuple consisting two new individuals.
+        """
+        child_1 = ChildIndividual(deepcopy(father1.neural_network))
+        child_2 = ChildIndividual(deepcopy(father2.neural_network))
+        for _ in range(weights_to_be_swapped):
+            random_layer_index = child_1.get_random_layer_index()
+            child_1_rand_layer = child_1.neural_network.hidden_layers[
+                random_layer_index
+            ]
+            child_2_rand_layer = child_2.neural_network.hidden_layers[
+                random_layer_index
+            ]
+            weight_index = ChildIndividual.get_random_weight_index(child_1_rand_layer)
+            (
+                child_1_rand_layer.weights[weight_index],
+                child_2_rand_layer.weights[weight_index],
+            ) = (
+                child_2_rand_layer.weights[weight_index],
+                child_1_rand_layer.weights[weight_index],
+            )
+        return child_1, child_2
+
+    @staticmethod
+    def neuron_swap_reproduction(
+        mother1: "Individual", mother2: "Individual", neurons_to_swap: int = 1
+    ) -> Tuple[ChildIndividual, ChildIndividual]:
+        """
+        Return two new individuals (children). They are created by
+        swapping one or more randomly chosen neurons.
+
+        :param mother1: First parent
+        :param mother2: Second parent
+        :param neurons_to_swap: Number of neurons to be swapped
+        :return: Tuple consisting two new individuals.
+        """
+        child_1 = ChildIndividual(deepcopy(mother1.neural_network))
+        child_2 = ChildIndividual(deepcopy(mother2.neural_network))
+        for _ in range(neurons_to_swap):
+            random_layer_index = child_1.get_random_layer_index()
+            child_1_rand_layer_w = child_1.neural_network.hidden_layers[
+                random_layer_index
+            ].weights
+            child_2_rand_layer_w = child_2.neural_network.hidden_layers[
+                random_layer_index
+            ].weights
+            neuron_index = np.random.randint(0, child_2_rand_layer_w.shape[0])
+            child_1_rand_layer_w[neuron_index], child_2_rand_layer_w[neuron_index] = (
+                child_2_rand_layer_w[neuron_index],
+                child_1_rand_layer_w[neuron_index].copy(),
```

Comment:
You already know. While testing it turned out that only one child had its neuron changed, so this copy() is necessary.

Reply:
Even though I may already know this, it should be documented in the code, as it is surprising. It may possibly be better to put it in a utility function which would operate on `numpy` arrays, swapping whole rows/columns.

Reply:
Done